src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/MathFunction.cs

```diff
@@ -67,23 +68,172 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
 
             var arg = args[0];
             var argType = argTypes[0];
-            fValid &= CheckNumericColumnType(argType, arg, errors, ref nodeToCoercedTypeMap);
+            fValid &= CheckNumericColumnType(argType, arg, errors, ref nodeToCoercedTypeMap, context, out returnType);
 
-            if (nodeToCoercedTypeMap?.Any() ?? false)
-            {
-                // Now set the coerced type to a table with numeric column type with the same name as in the argument.
-                returnType = nodeToCoercedTypeMap[arg];
-            }
-            else
+            if (!fValid)
             {
-                returnType = argType;
+                nodeToCoercedTypeMap = null;
             }
 
-            returnType = context.Features.ConsistentOneColumnTableResult ? DType.CreateTable(new TypedName(DType.Number, GetOneColumnTableResultName(context.Features))) : returnType;
+            return fValid;
+        }
+    }
+
+    // Abstract base class for all 2-arg math functions that return numeric values.
+    internal abstract class MathTwoArgFunction : BuiltinFunction
+    {
+        public override ArgPreprocessor GetArgPreprocessor(int index)
+        {
+            return base.GetGenericArgPreprocessor(index);
+        }
 
-            if (!fValid)
+        public override bool IsSelfContained => true;
+
+        public MathTwoArgFunction(string name, TexlStrings.StringGetter description, int minArity)
+            : base(name, description, FunctionCategories.MathAndStat, DType.Number, 0, minArity, 2, DType.Number, DType.Number)
+        {
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            if (MinArity == 1)
+            {
+                yield return new[] { TexlStrings.MathTFuncArg1 };
+            }
+
+            yield return new[] { TexlStrings.MathTFuncArg1, TexlStrings.MathTFuncArg2 };
+        }
+    }
+
+    internal abstract class MathTwoArgTableFunction : BuiltinFunction
+    {
+        public override bool IsSelfContained => true;
+
+        // Before ConsistentOneColumnTableResult, this function would always return a fixed name "Result" (Mod)
+        protected virtual bool InConsistentTableResultFixedName => false;
+
+        // Before ConsistentOneColumnTableResult, this function would use the second argument name if a table (Log, Power)
+        protected virtual bool InConsistentTableResultUseSecondArg => false;
+
+        public MathTwoArgTableFunction(string name, TexlStrings.StringGetter description, int minArity)
```

Comment:
nit: Can you use the explicit parameter name syntax: minArity: 1 etc, in all the places where this constructor is called?

Reply:
Done.  Good call.